Source/BoundaryConditions/PML.cpp

```diff
@@ -1045,143 +1051,66 @@ PML::ComputePMLFactors (amrex::Real dt)
     }
 }
 
-std::array<MultiFab*,3>
-PML::GetE_fp ()
-{
-    return {pml_E_fp[0].get(), pml_E_fp[1].get(), pml_E_fp[2].get()};
-}
-
-std::array<MultiFab*,3>
-PML::GetB_fp ()
-{
-    return {pml_B_fp[0].get(), pml_B_fp[1].get(), pml_B_fp[2].get()};
-}
-
-std::array<MultiFab*,3>
-PML::Getj_fp ()
-{
-    return {pml_j_fp[0].get(), pml_j_fp[1].get(), pml_j_fp[2].get()};
-}
-
-std::array<MultiFab*,3>
-PML::GetE_cp ()
-{
-    return {pml_E_cp[0].get(), pml_E_cp[1].get(), pml_E_cp[2].get()};
-}
-
-std::array<MultiFab*,3>
-PML::GetB_cp ()
-{
-    return {pml_B_cp[0].get(), pml_B_cp[1].get(), pml_B_cp[2].get()};
-}
-
-std::array<MultiFab*,3>
-PML::Getj_cp ()
-{
-    return {pml_j_cp[0].get(), pml_j_cp[1].get(), pml_j_cp[2].get()};
-}
-
-std::array<MultiFab*,3>
-PML::Get_edge_lengths()
-{
-    return {pml_edge_lengths[0].get(), pml_edge_lengths[1].get(), pml_edge_lengths[2].get()};
-}
-
-
-MultiFab*
-PML::GetF_fp ()
-{
-    return pml_F_fp.get();
-}
-
-MultiFab*
-PML::GetF_cp ()
-{
-    return pml_F_cp.get();
-}
-
-MultiFab*
-PML::GetG_fp ()
+void
+PML::CopyJtoPMLs (
+    ablastr::fields::MultiFabRegister& fields,
+    PatchType patch_type,
+    int lev
+)
 {
-    return pml_G_fp.get();
-}
+    using ablastr::fields::Direction;
 
-MultiFab*
-PML::GetG_cp ()
-{
-    return pml_G_cp.get();
-}
+    bool const has_j_fp = fields.has(FieldType::current_fp, Direction{0}, lev);
+    bool const has_pml_j_fp = fields.has(FieldType::pml_j_fp, Direction{0}, lev);
+    bool const has_j_cp = fields.has(FieldType::current_cp, Direction{0}, lev);
+    bool const has_pml_j_cp = fields.has(FieldType::pml_j_cp, Direction{0}, lev);
```

Comment:
These could probably be replaced by calls to `has_vector`, which could be a bit more robust maybe?
```suggestion
    bool const has_j_fp = fields.has_vector(FieldType::current_fp, lev);
    bool const has_pml_j_fp = fields.has_vector(FieldType::pml_j_fp, lev);
    bool const has_j_cp = fields.has_vector(FieldType::current_cp, lev);
    bool const has_pml_j_cp = fields.has_vector(FieldType::pml_j_cp, lev);
```

Reply:
Since `has_vector` was introduced later, there are a lot of instances where `has` had been used (fixing a direction index, say `0`) and could be replaced by `has_vector`. Basically most of the times that `has` is called for vectors. This could a separate follow-up PR.

Reply:
Agreed. Let's add those above and then do more in a follow-up.